src/source/put.js

```diff
@@ -49,6 +44,26 @@ function getUser(context) {
   return email || 'anonymous';
 }
 
+/**
+ * Put file based on path and body in the source bus.
+ *
+ * @param {import('../support/AdminContext').AdminContext} context context
+ * @param {string} path path to store the file (including extension)
+ * @param {string} mime the mime type of the file
+ * @param {Buffer} body content body
+ * @returns {Promise<Response>} response
+ */
+export async function putSourceFile(context, path, mime, body) {
```

Comment:
since the path is used as key in for the bucket, I would name it `key`

```suggestion
export async function putSourceFile(context, key, mime, body) {
```